app/Filament/Resources/GameResource/Pages/Hubs.php

```diff
@@ -96,20 +105,44 @@ public function table(Table $table): Table
                                     ->limit(50)
                                     ->get()
                                     ->mapWithKeys(fn ($hub) => [$hub->id => "[{$hub->id}] {$hub->title}"]);
-                            }),
+                            })
+                            ->hidden(fn (Forms\Get $get): bool => filled($get('hub_ids_csv')))
+                            ->disabled(fn (Forms\Get $get): bool => filled($get('hub_ids_csv')))
+                            ->live()
+                            ->helperText('... or search and select hubs to add.'),
                     ])
                     ->modalHeading('Add hubs to game')
                     ->modalAutofocus(false)
                     ->action(function (array $data): void {
                         /** @var Game $game */
                         $game = $this->getOwnerRecord();
 
-                        $gameSets = GameSet::whereType(GameSetType::Hub)
-                            ->whereIn('id', $data['hub_ids'])
-                            ->get();
+                        // Handle select field input.
+                        if (!empty($data['hub_ids'])) {
+                            $gameSets = GameSet::whereType(GameSetType::Hub)
+                                ->whereIn('id', $data['hub_ids'])
+                                ->get();
 
-                        foreach ($gameSets as $gameSet) {
-                            (new AttachGamesToGameSetAction())->execute($gameSet, [$game->id]);
+                            foreach ($gameSets as $gameSet) {
+                                (new AttachGamesToGameSetAction())->execute($gameSet, [$game->id]);
+                            }
+
+                            return;
+                        }
+
+                        // Handle CSV input.
+                        if (!empty($data['hub_ids_csv'])) {
+                            $hubIds = (new ParseIdsFromCsvAction())->execute($data['hub_ids_csv']);
+
+                            // Validate that these hubs can be attached.
+                            $gameSets = GameSet::whereType(GameSetType::Hub)
+                                ->whereIn('id', $hubIds)
+                                ->whereNotIn('id', $this->getOwnerRecord()->hubs->pluck('id'))
+                                ->get();
```

Comment:
Would it be worth having feedback for invalid IDs? I thought this wasn't working when I tried to add a couple random IDs that turned out to be similar-games game sets.

Reply:
I did manage to get a working implementation of this, but it's a _lot_ of code. The complexity is mostly around what happens for the mixed case where some of the IDs are valid and others are not.

Let's go with the current implementation in the branch, and if this becomes a broader point of confusion, maybe we can revisit.